src/MOI_wrapper/MOI_wrapper.jl

```diff
@@ -220,15 +220,31 @@ _bounds(s::MOI.LessThan{Float64}) = (-Inf, s.upper)
 _bounds(s::MOI.EqualTo{Float64}) = (s.value, s.value)
 _bounds(s::MOI.Interval{Float64}) = (s.lower, s.upper)
 
-function _extract_row_data(src, mapping, lb, ub, I, J, V, S)
+function add_sizehint!(vec, n)
+    len = length(vec)
+    return sizehint!(vec, len + n)
+end
+
+function _extract_row_data(src, mapping, lb, ub, I, J, V, ::Type{S}) where S
     row = length(I) == 0 ? 1 : I[end] + 1
-    for c_index in MOI.get(
+    list = MOI.get(
         src, MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{Float64}, S}()
     )
+    add_sizehint!(lb, length(list))
+    add_sizehint!(ub, length(list))
+    n_terms = 0
+    for c_index in list
         f = MOI.get(src, MOI.ConstraintFunction(), c_index)
         l, u = _bounds(MOI.get(src, MOI.ConstraintSet(), c_index))
         push!(lb, l - f.constant)
         push!(ub, u - f.constant)
+        n_terms += length(f.terms)
+    end
+    add_sizehint!(I, n_terms)
+    add_sizehint!(J, n_terms)
+    add_sizehint!(V, n_terms)
+    for c_index in list
```

Comment:
Why are we doing the loop twice? Is it for cache friendliness as we modify different vectors?
The disadvantage is that we get the function twice

Reply:
I agree. How much does this particular optimization help?

Reply:
Doing twice we can pre allocate slots in I, J and V.
It does not seem to make the code any more readable or unreadable.
It is responsible for 30% of the overall speedup.
An extra step would be to cache f also, I was thinking it might be way to much, but that can give and extra 10%:
```
 ──────────────────────────────────────────────────────────────────
                           Time                   Allocations
                   ──────────────────────   ───────────────────────
 Tot / % measured:      52.8s / 100%            8.91GiB / 100%

 Section   ncalls     time   %tot     avg     alloc   %tot      avg
 ──────────────────────────────────────────────────────────────────
 bcs + v       10    10.9s  20.7%   1.09s   2.11GiB  23.7%   216MiB
   opt         10    10.2s  19.3%   1.02s    938MiB  10.3%  93.8MiB
   build       10    744ms  1.41%  74.4ms   1.19GiB  13.4%   122MiB
 bcs           10    10.6s  20.0%   1.06s   1.58GiB  17.7%   161MiB
   opt         10    10.1s  19.1%   1.01s    938MiB  10.3%  93.8MiB
   build       10    454ms  0.86%  45.4ms    677MiB  7.43%  67.7MiB
 bc + s        10    10.3s  19.4%   1.03s   1.53GiB  17.1%   156MiB
   opt         10    8.31s  15.7%   831ms     0.00B  0.00%    0.00B
   copy        10    1.72s  3.25%   172ms    887MiB  9.73%  88.7MiB
   build       10    212ms  0.40%  21.2ms    676MiB  7.41%  67.6MiB
 cs            10    10.1s  19.2%   1.01s   1.58GiB  17.7%   161MiB
   opt         10    9.80s  18.6%   980ms    938MiB  10.3%  93.8MiB
   build       10    332ms  0.63%  33.2ms    677MiB  7.43%  67.7MiB
 c + s         10    10.1s  19.1%   1.01s   1.51GiB  17.0%   155MiB
   opt         10    8.15s  15.4%   815ms     0.00B  0.00%    0.00B
   copy        10    1.62s  3.07%   162ms    874MiB  9.58%  87.4MiB
   build       10    325ms  0.62%  32.5ms    676MiB  7.41%  67.6MiB
 data          10    789ms  1.49%  78.9ms    617MiB  6.76%  61.7MiB
 ──────────────────────────────────────────────────────────────────
```

Reply:
Actually, its VERY important for the bridged cache with separate solver case!
I will commit it!

Reply:
Now bridging overhead is basically zero, and copy is less than 20% fo solve time

Reply:
Indeed, I missed the computation of `n_terms`, it makes sense then

Reply:
Why is getting functions slow ? Isn't the bridged cache just a `MOIU.AbstractModel` ?

Reply:
not sure why, yes it should be.

Reply:
I don't think its specially slow though.
The loop is just tight.